pkg/backuphelpers/backupvars_test.go

```diff
@@ -0,0 +1,113 @@
+package backuphelpers
+
+import (
+	"testing"
+
+	backupv1alpha1 "github.com/openshift/api/config/v1alpha1"
+	prune "github.com/openshift/cluster-etcd-operator/pkg/cmd/prune-backups"
+
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	schedule = "0 */2 * * *"
+	timezone = "GMT"
+)
+
+func TestBackupConfig_ToArgs(t *testing.T) {
+	testCases := []struct {
+		name     string
+		cr       *backupv1alpha1.EtcdBackupSpec
+		expected string
+	}{
+		{
+			"enabled",
+			createEtcdBackupSpec(timezone, schedule),
+			"    args:\n    - --enabled=true\n    - --timezone=GMT\n    - --schedule=0 */2 * * *",
+		},
+		{
+			"disabled",
+			createEtcdBackupSpec(timezone, schedule),
+			"    args:\n    - --enabled=true\n    - --timezone=GMT\n    - --schedule=0 */2 * * *",
+		},
+		{
+			"empty schedule disabled",
+			createEtcdBackupSpec(timezone, ""),
+			"    args:\n    - --enabled=true\n    - --timezone=GMT",
+		},
+		{
+			"empty schedule enabled",
+			createEtcdBackupSpec(timezone, ""),
+			"    args:\n    - --enabled=true\n    - --timezone=GMT",
+		},
+
+		{
+			"empty timezone disabled",
+			createEtcdBackupSpec("", schedule),
+			"    args:\n    - --enabled=true\n    - --schedule=0 */2 * * *",
+		},
+		{
+			"empty timezone enabled",
+			createEtcdBackupSpec("", schedule),
+			"    args:\n    - --enabled=true\n    - --schedule=0 */2 * * *",
+		},
+		{
+			"empty timezone and schedule disabled",
+			createEtcdBackupSpec("", ""),
+			"    args:\n    - --enabled=false",
+		},
+		{
+			"empty timezone and schedule enabled",
+			createEtcdBackupSpec("", ""),
+			"    args:\n    - --enabled=false",
+		},
+		{
+			"retention number",
+			withRetentionNumberThreeBackups(createEtcdBackupSpec(timezone, schedule)),
+			"    args:\n    - --enabled=true\n    - --timezone=GMT\n    - --schedule=0 */2 * * *\n    - --type=RetentionNumber\n    - --maxNumberOfBackups=3",
+		},
+		{
+			"retention none",
+			withRetentionSizeOneGB(createEtcdBackupSpec(timezone, schedule)),
+			"    args:\n    - --enabled=true\n    - --timezone=GMT\n    - --schedule=0 */2 * * *\n    - --type=RetentionSize\n    - --maxSizeOfBackupsGb=1",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+
+			c := NewDisabledBackupConfig()
+			c.SetBackupSpec(tc.cr)
+
+			act := c.ArgString()
+
+			require.Equal(t, tc.expected, act)
+		})
+	}
+}
+
+func createEtcdBackupSpec(timezone, schedule string) *backupv1alpha1.EtcdBackupSpec {
+	if timezone == "" && schedule == "" {
+		return nil
+	}
```

Comment:
ah okay that's how I got confused. Can you please remove this and just explicitly set "nil" in the disabled cases above?

Reply:
but i can not create a Spec without `schedule`, or ?

Reply:
done, but this way we can have a `BackupSpec` without `schedule`, i think this is not correct !!

Reply:
that's the purpose of a unit test, isn't it? ;)

Reply:
:D :D .. yes i know, but i think we should `not` enable the container if the schedule is empty, or ?

Reply:
you can't create such a CR because it's a required field in the apiserver

Reply:
ahhh .. so it will be rejected by the APIServer  .. alright thats a relief

Reply:
apparently it's not, but it's certainly something we should add for a new version of the CRD then:
https://github.com/openshift/api/blob/master/config/v1alpha1/types_backup.go#L51

Then again "Empty string means no opinion and the platform is left to choose a reasonable default which is subject to change without notice." feel free to set something by default in the next PR in the sidecar logic

Reply:
lgtm 👍🏽